e2e/_suites/ingest-manager/stand-alone.go

```diff
@@ -66,13 +81,156 @@ func (sats *StandAloneTestSuite) aStandaloneAgentIsDeployed() error {
 }
 
 func (sats *StandAloneTestSuite) thereIsNewDataInTheIndexFromAgent() error {
-	return godog.ErrPending
+	maxTimeout := time.Duration(queryRetryTimeout) * time.Minute
+	minimumHitsCount := 100
+
+	result, err := searchAgentData(sats.Hostname, sats.RuntimeDependenciesStartDate, minimumHitsCount, maxTimeout)
+	if err != nil {
+		return err
+	}
+
+	log.Debugf("Search result: %v", result)
+
+	return e2e.AssertHitsArePresent(result)
 }
 
-func (sats *StandAloneTestSuite) theDockerContainerIsStopped(arg1 string) error {
-	return godog.ErrPending
+func (sats *StandAloneTestSuite) theDockerContainerIsStopped(serviceName string) error {
+	serviceManager := services.NewServiceManager()
+
+	err := serviceManager.RemoveServicesFromCompose("ingest-manager", []string{serviceName}, profileEnv)
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":   err,
+			"service": serviceName,
+		}).Error("Could not stop the service.")
+
+		return err
+	}
+	sats.AgentStoppedDate = time.Now()
+
+	return nil
 }
 
 func (sats *StandAloneTestSuite) thereIsNoNewDataInTheIndexAfterAgentShutsDown() error {
-	return godog.ErrPending
+	maxTimeout := time.Duration(30) * time.Second
+	minimumHitsCount := 1
+
+	result, err := searchAgentData(sats.Hostname, sats.AgentStoppedDate, minimumHitsCount, maxTimeout)
```

Comment:
We use the moment the agent stopped to check for documents

Reply:
I should have caught this earlier, but indeed - we should put in some minimal (1 second or 2??) padding on the chance that a doc was sent right as agent is stopping and is in ES microseconds after the agent stop time is recorded  @mdelapenya what do you think?

Reply:
Flakiness at sight!!

TBH not sure, we can add a few seconds to that time

Reply:
Let's merge as is, so we can observe how it behaves